### PR TITLE
New doc

### PR DESCRIPTION
**Pull request by M0ses**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
+2. [How to contribute code](#contribute_code)
+3. [How to contribute issues](#contribute_issues)
+4. [Contribute to the OBS documentation](#contribute_docu)
+5. [Conduct](#conduct)
+6. [Communication](#communication)
+7. [Rubocop](#rubocop)
+8. [Setup an OBS backend for development](#setup_backend)
+9. [Quick Start Guid (Howto setup a developer VM)](#quick_start)
+
+# <a name="request"/> Request for contributions
```
Comment (hennevogel) [inline CONTRIBUTING.md]: Why do you introduce those html anchors?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -67,3 +79,74 @@ If you want to take part of this please follow this process:
 * Create a commit with all changes.
 
 * Go to the next rubocop offense.
+
+# <a name="setup_backend"/>Setup an OBS backend for development
```
Comment (hennevogel) [inline CONTRIBUTING.md]: Vagrant includes the instructions on how to run a backend right? Why duplicate this?

File: README.md [hunk 3]
```diff
@@ -27,326 +27,36 @@ The OBS source code repository is hosted on [Github](http://github.com/opensuse/
 ## Setup
 There are 3 scenarios for which you can setup an OBS instance. Running it in *production* for your users, for *development* on it and for executing the *test* suite.
```
Comment (hennevogel) [inline README.md]: Drop the *test suite* case, this is covered in CONTRIBUTING.md

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -67,3 +79,74 @@ If you want to take part of this please follow this process:
...
+
+1. Install [Vagrant](https://www.vagrantup.com/downloads.html) and [VirtualBox](https://www.virtualbox.org/wiki/Downloads). Both tools support Linux, MacOS and Windows and in principal setting up your OBS development environment works similar.
+
+2. Install [vagrant-exec](https://github.com/p0deje/vagrant-exec):
+
+    ```
+    vagrant plugin install vagrant-exec
+    vagrant plugin install vagrant-reload
+    # optional if you are running vagrant with libvirt (e.g. kvm)
+    vagrant plugin install vagrant-libvirt
```
Comment (hennevogel) [inline CONTRIBUTING.md]: I would skip this as people following the rest of the README won't need it...

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -67,3 +79,74 @@ If you want to take part of this please follow this process:
...
+Check [src/backend/README](https://github.com/openSUSE/open-build-service/blob/master/src/backend/README) how to run the backend from the source code repository.
+
+
+# <a name="quick_start"/>Quick Start Guid (Howto setup a developer VM)
+
+We are using [Vagrant](https://www.vagrantup.com/) to create our development environments.
+
+1. Install [Vagrant](https://www.vagrantup.com/downloads.html) and [VirtualBox](https://www.virtualbox.org/wiki/Downloads). Both tools support Linux, MacOS and Windows and in principal setting up your OBS development environment works similar.
+
+2. Install [vagrant-exec](https://github.com/p0deje/vagrant-exec):
```
Comment (hennevogel) [inline CONTRIBUTING.md]: +2. Install Vagrant plugins

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -67,3 +79,74 @@ If you want to take part of this please follow this process:
 * Create a commit with all changes.
 
 * Go to the next rubocop offense.
+
+# <a name="setup_backend"/>Setup an OBS backend for development
+
+Check [src/backend/README](https://github.com/openSUSE/open-build-service/blob/master/src/backend/README) how to run the backend from the source code repository.
+
+
+# <a name="quick_start"/>Quick Start Guid (Howto setup a developer VM)
```
Comment (hennevogel) [inline CONTRIBUTING.md]: *Guide*

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -28,22 +40,22 @@ We are using priority labels from **P0** to **P4** for our issues. So if you are
 * P3: Medium - Fix this when you have time.
 * P4: Low  - Fix this when you don't see any issues with the other priorities.
 
-# Contribute to the OBS documentation
+# <a name="contribute_docu" />Contribute to the OBS documentation
```
Comment (hennevogel) [inline CONTRIBUTING.md]: +How to contribute documentation

File: dist/README.SETUP [hunk 8]
```diff
@@ -1,45 +1,49 @@
 # Manual Setup Guide
```
Comment (hennevogel) [inline dist/README.SETUP]: Why do we have both? README.SETUP.md and README.SETUP

**Comment by hennevogel** [comment]
In general README.SETUP.md is exactly what I wanted to avoid:  The only content that has changed for our users is that we have introduced a 5th method of installing OBS (setup-appliance.sh)...

File: CONTRIBUTING.md (hunk 1 above)
Comment (M0ses) [inline CONTRIBUTING.md]: sure ?

https://github.com/M0ses/testing_markdown/blob/master/README.md#Test%20for%20TOC

Comment (hennevogel) [inline CONTRIBUTING.md]: If you move your mouse left of the title you see the real anchor. It's lowercase and with hyphens :-)

File: dist/README.SETUP (hunk 8 above)
Comment (M0ses) [inline dist/README.SETUP]: * README.SETUP.md for nice view on github
* link README.SETUP for "backwards compatiblity"

File: CONTRIBUTING.md (hunk 6 above)
Comment (M0ses) [inline CONTRIBUTING.md]: THX

File: CONTRIBUTING.md (hunk 2 above)
Comment (adrianschroeter) [inline CONTRIBUTING.md]: You may develop also in backend parts and esp. with distributed setup. So we need some place somewhere how the backend gets setup.

File: README.md (hunk 3 above)
Comment (adrianschroeter) [inline README.md]: Actually, I would also drop development here, just pointing developers to the CONTRIBUTING file.
Just saying "production" is what you want.

File: CONTRIBUTING.md (hunk 2 above)
Comment (M0ses) [inline CONTRIBUTING.md]: No, it's not the same , e.g.  start_test_backend
* does not start workers
* runs bs_publish in testmode 
* etc.

**Comment by M0ses** [comment]
@hennevogel : 

* I removed redundant documentation (README.md/README.SETUP)
* I added recommendations when to use which way
* The README.md became much clearer

And no, we did not introduced the 5th way - There are 3 ways of installation with clear recommendations and example which one to use to achive a specific goal.

File: CONTRIBUTING.md (hunk 1 above)
Comment (M0ses) [inline CONTRIBUTING.md]: @hennevogel  THX, will fix this

File: dist/README.SETUP (hunk 8 above)
Comment (hennevogel) [inline dist/README.SETUP]: But README.SETUP is is not a link...

File: CONTRIBUTING.md (hunk 2 above)
Comment (hennevogel) [inline CONTRIBUTING.md]: Then you have to make clear what to do when. Otherwise people will be confused.

**Comment by hennevogel** [comment]
@M0ses The content (not the structure or location files etc.) before was

* install via appliance
* install via packages

Now the content is 

* install via appliance
* install via setup-appliance.sh
* install via packages

**Comment by hennevogel** [comment]
So what are we going to do about this?

**Comment by hennevogel** [comment]
Talked to @M0ses offline. We're tracking the missing things in [trello](https://trello.com/c/ot4mIxke)

**Pull request merged**